src/resource.coffee

```diff
@@ -91,14 +91,14 @@ class Resource extends Model
     else
       throw new Error 'Can\'t uncache a resource with no ID'
 
-  delete: ->
+  delete: (query={})->
     @_write = @_write
       .catch =>
         null
       .then =>
         deletion = if @id
           @refresh(true, query).then =>
-            @_type._client.delete @_getURL(), null, @_getHeadersForModification()
+            @_type._client.delete @_getURL(), query, @_getHeadersForModification()
```

Comment:
Is the query param converting to the payload in the delete request here? Noting that delete HTTP reqs have no payload, so it's doing nothing. 

https://github.com/zooniverse/json-api-client/blob/48426c58de25073142878d81f69ea5383f7f10c2/src/json-api-client.coffee#L62

https://github.com/zooniverse/json-api-client/blob/48426c58de25073142878d81f69ea5383f7f10c2/src/json-api-client.coffee#L33

Should the order be `@_type._client.delete @_getURL(), null, @_getHeadersForModification(), query` instead?

Reply:
The payload for delete requests gets passed to the API as query parameters in #39. Looking at that PR, the `query` argument is ignored by `.delete`. This is confusing!

Reply:
I remember that the last argument to `makeHTTPRequest` was a hack, because POST and PUT requests can have both a payload and query params. Prior to that change, the client didn't distinguish between payload and query params.

Reply:
right - if superagent can do `request.delete(url).params(data).params(params)` then i think we can match the method arguments correctly and not push the params into the payload here.